hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -194,6 +202,47 @@ func (r *HostedClusterReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 		}
 	}
 
+	// Set the ignition server availability condition by checking its deployment.
+	{
+		controlPlaneNamespace := manifests.HostedControlPlaneNamespace(hcluster.Namespace, hcluster.Name)
+		deployment := ignitionserver.Deployment(controlPlaneNamespace.Name)
+		if err := r.Client.Get(ctx, client.ObjectKeyFromObject(deployment), deployment); err != nil {
+			if apierrors.IsNotFound(err) {
+				meta.SetStatusCondition(&hcluster.Status.Conditions, metav1.Condition{
+					Type:               string(hyperv1.IgnitionServerAvailable),
+					Status:             metav1.ConditionFalse,
+					ObservedGeneration: hcluster.Generation,
+					Reason:             "IgnitionServerDeploymentNotFound",
+				})
+			} else {
+				return ctrl.Result{}, fmt.Errorf("failed to get ignition server deployment: %w", err)
+			}
+		} else {
+			available := false
+			for _, cond := range deployment.Status.Conditions {
+				if cond.Type == appsv1.DeploymentAvailable && cond.Status == corev1.ConditionTrue {
+					available = true
+					break
+				}
+			}
+			if available {
+				meta.SetStatusCondition(&hcluster.Status.Conditions, metav1.Condition{
+					Type:               string(hyperv1.IgnitionServerAvailable),
+					Status:             metav1.ConditionTrue,
+					ObservedGeneration: hcluster.Generation,
+					Reason:             "IgnitionServerDeploymentAvailable",
+				})
+			} else {
+				meta.SetStatusCondition(&hcluster.Status.Conditions, metav1.Condition{
+					Type:               string(hyperv1.IgnitionServerAvailable),
+					Status:             metav1.ConditionFalse,
+					ObservedGeneration: hcluster.Generation,
+					Reason:             "IgnitionServerDeploymentUnavailable",
+				})
+			}
+		}
+	}
```

Comment:
we are setting this now but not taking it into consideration? Should the nodePool_controller line 211 add a check for this condition in addition to `Status.IgnitionEndpoint`?

Reply:
Correct, it's not being read anywhere yet in this PR. Thinking about this... does ignition server availability impact our ability to reconcile nodepools here? The required secrets may exist but the server's availability could flap; either way we can still reconcile. I think as long as the secrets exist we can make progress; if the ign server itself isn't available at any time (e.g. the time instances actually connect to it), that's a separate problem to solve for.